design-system/react/src/components/ButtonLink/defs.ts

```diff
@@ -0,0 +1,29 @@
+import type { ButtonProps } from '../Button/defs';
+
+import type { Components } from '~/defs';
+import type { CreateComponent, HTMLProps } from '~/utils';
+
+type PropsToOmit =
+  | 'as'
```

Comment:
this component should not omit the `as`, it can be a polymorphic one, since you can use other link components as nextjs `Link` or so